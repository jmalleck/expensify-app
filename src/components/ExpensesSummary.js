import React from 'react';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';
import { connect } from 'react-redux';

export const ExpensesSummary = ({ numOfExpenses, expensesTotal }) => (
    <div>
        <h1>
            Viewing {numOfExpenses}
            {numOfExpenses === 1 ? <span> expense </span> : <span> expenses </span>}
            totalling {numeral(expensesTotal / 100).format('$0,0.00')}
        </h1>
    </div>
);

const mapStateToProps = (state) => {
    const filteredExpenses = (selectExpenses(state.expenses, state.filters));
    return {
        numOfExpenses: filteredExpenses.length,
        expensesTotal: selectExpensesTotal(filteredExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);