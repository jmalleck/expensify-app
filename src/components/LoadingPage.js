import React from 'react';

const LoadingPage = () => (
    <div className="loader">
        <img className="loader__image" src="/img/loader.gif"></img>
    </div>
);

export default LoadingPage;