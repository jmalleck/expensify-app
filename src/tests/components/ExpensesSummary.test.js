import { shallow } from 'enzyme';
import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';


test('Should display ExpensesSummary correctly with no expenses', () => {
    const wrapper = shallow(<ExpensesSummary />);
    expect(wrapper).toMatchSnapshot();
});

test('Should display ExpensesSummary correctly with expenses', () => {
    const wrapper = shallow(<ExpensesSummary numOfExpenses={2} expensesTotal={150} />);
    expect(wrapper).toMatchSnapshot();
});