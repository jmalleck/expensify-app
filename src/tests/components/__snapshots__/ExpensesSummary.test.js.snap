// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display ExpensesSummary correctly with expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
    <span>
       expenses 
    </span>
    totalling 
    $1.50
  </h1>
</div>
`;

exports[`Should display ExpensesSummary correctly with no expenses 1`] = `
<div>
  <h1>
    Viewing 
    <span>
       expenses 
    </span>
    totalling 
    $0.00
  </h1>
</div>
`;
